pages/search_section.py

```diff
@@ -33,11 +33,12 @@ def mouse_over_cart_icon(self):
         action = ActionChains(self.driver)
         action.move_to_element(cartIcon).perform()
 
-    def cart_popup_product_number(self):
-        return self.get_element(by_locator=self.__CART_POPUP_PRODUCT_NUMBER)
+    def check_cart_popup_product_number(self, expected_product_number: int) -> bool:
+        element_locator = (By.XPATH, self.__CART_POPUP_PRODUCT_NUMBER.format(index=expected_product_number))
+        self.if_element_present(by_locator=element_locator)
```

Comment:
format on tuple
missing return

I'd like to point this:
```python
    def cart_icon(self):
        return self.get_element(by_locator=self.__CART_ICON)
        
    def mouse_over_cart_icon(self):
        cartIcon = self.cart_icon()
        action = ActionChains(self.driver)
        action.move_to_element(cartIcon).perform()
```
if you move it to the base class like I mentioned you a few comments before, you'll have a case to reuse that method.
also you have a `get_element` method in a base class so you don't need to create `cart_icon` method here